src/marker.rs

```diff
@@ -14,105 +14,81 @@
 //! awaiting a future
 //! - Once that is done, reacquire the GIL
 //!
-//! That API is provided by [`Python::allow_threads`] and enforced via the [`Ungil`] bound on the
-//! closure and the return type. This is done by relying on the [`Send`] auto trait. `Ungil` is
-//! defined as the following:
```

Comment:
> As per discussion on https://github.com/PyO3/pyo3/issues/3658 it seems to me that in the long-term it might still be the better auto trait than Send (once GILProtected is removed and nogil is the only build mode).

I think when that day actually comes, this API itself might be on the chopping block. Until then, I actually feel like `Send` is somewhat appropriate as we do talk about state (holding the GIL) attached to the current thread.

I certainly like the reduction of accidental complexity (auto trait that actually is `Send` on stable, features that only work on nightly, etc.).